### ISSUE TITLE
Add Document link to Frontend Chat App

### ISSUE THREAD

**Issue by sekasx** [ctrl-space-labs/gendox-core#220]
### Description

Links of document sections need to be added in the chat app similar to the ones already exist in Discord.
The link should open the document in the Frontend and navigate to the appropriate section.

<img width="1122" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMGxGQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--fae05cc1628fb48b1e6f5601575ccbb7632e1527/image.png" alt="image.png" />

**Comment by gyfantidis** [comment]
- Will need to be stored somewhere in the database the `instanceSections`,  
- Now they are not stored anywhere just sent your reply to the `Semantic completion` request with the `message`
``` 
CompletionMessageDTO completionMessageDTO = CompletionMessageDTO.builder()
                .message(completion)
                .sectionId(instanceSections.stream().map(DocumentInstanceSection::getId).toList())
                .threadID(message.getThreadId())
                .build();

 ```

- In case we want it to appear only in the last message of our chat, it is enough, but if we want it to appear in the previous ones as well, we should get them from the database,

- I suggest that to create a new column in `message` table, with name `closestSections` .
- @sekasx

**Comment by sekasx** [comment]
@gyfantidis Good point! 1 column is not enough, the number of sections included in messages is more than 1, so it's many-to-many. 
Also ,we have to think about how this will work along with ProvenAI, where the Sections can come from other sources that are not stored in our DB.

For now, let's create a new pivot table `message_section` with columns:
- message_id
- section_id 
- document_id
- section_iscc
- sections_url

in the `Message` Entity, a `List<MessageSection> messageSections` can be added. In the Frontend, you only need the `section_url`. The URL can be constructed either in the backend or in Frontend (in this case will be null in the DB)

When the user clicks the link, the document page will be opened and auto-scroll to the section.

------
@myrtp with ProvenAI this table will be expanded to store the information provided in the PermissionOfUse VC. Let's discuss it and open a new ticket for this.
